File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    // understand the problem
    // outline the input and output
    // explain your approach
    // pseudocode 
    // code
        
    for(let i = 0; i < nums.length; i++){
        for(let j = i + 1; j < nums.length; j++){
            if(nums[i] + nums[j] == target){
                return [i, j]
            }
        }
    }

};

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        nums_map = {}

        for i, number in enumerate(nums):
            nums_map[number] = i

        for i, number in enumerate(nums):
            number_needed = target - number #[3:0,2:1,4:2] -> 6 - 3 = 3
            #   3           [map] and indx(0!=0)
            if number_needed in nums_map and nums_map[number_needed] != i:
                return [i, nums_map[number_needed]]